### PR TITLE
Feature/174706879 filter runlogs by contract address

### PR DESCRIPTION
**Pull request by j16r**

### CODE REVIEW

File: core/cmd/remote_client_test.go [hunk 1]
```diff
@@ -684,7 +684,7 @@ func TestClient_RemoteLogin(t *testing.T) {
 
 func setupWithdrawalsApplication(t *testing.T, config *cltest.TestConfig) (*cltest.TestApplication, func()) {
 	oca := common.HexToAddress("0xDEADB3333333F")
-	config.Set("ORACLE_CONTRACT_ADDRESS", &oca)
+	config.Set("OPERATOR_CONTRACT_ADDRESS", &oca)
```
Comment (samsondav) [inline core/cmd/remote_client_test.go:687]: Would it make sense to prefix with `ETH_`?

File: core/store/orm/config.go [hunk 2]
```diff
@@ -415,12 +415,12 @@ func (c Config) ExplorerSecret() string {
 	return c.viper.GetString(EnvVarName("ExplorerSecret"))
 }
 
-// OracleContractAddress represents the deployed Oracle contract's address.
-func (c Config) OracleContractAddress() *common.Address {
-	if c.viper.GetString(EnvVarName("OracleContractAddress")) == "" {
-		return nil
+// OperatorContractAddress represents the deployed Oracle contract's address.
```
Comment (samsondav) [inline core/store/orm/config.go:418]: Could we have a slightly more detailed explanation here?

**Review by samsondav — DISMISSED** [review]
LGTM with some minor points about documentation.

Also, could we add something to the CHANGELOG-core.md?

File: core/store/orm/config.go (hunk 2 above)
Comment (j16r) [reply core/store/orm/config.go:418]: @thodges-gh any suggestions on a more detailed explanation?

Comment (thodges-gh) [reply core/store/orm/config.go:418]: Operator.sol is the next version of the Oracle.sol contract that every node operator needs to deploy to accept RunLog requests.

File: core/cmd/remote_client_test.go (hunk 1 above)
Comment (j16r) [reply core/cmd/remote_client_test.go:687]: 🤷

**Comment by thodges-gh** [comment]
I get a panic when setting `OPERATOR_CONTRACT_ADDRESS` and starting the node:
```
panic: interface conversion: interface {} is *common.Address, not common.Address

goroutine 214 [running]:
github.com/smartcontractkit/chainlink/core/store/orm.Config.OperatorContractAddress(0xc0003de000, 0x185bc20, 0x210c6e8, 0xc0006ee880, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/thomas/workspace/chainlink/core/store/orm/config.go:424 +0x176
github.com/smartcontractkit/chainlink/core/services.NewInitiatorSubscription(0x1, 0xc0007ee9f0, 0xc0007ee81a, 0x6, 0x1e97c980, 0xed6dd24e6, 0xc0001aa8c0, 0x0, 0x0, 0x0, ...)
        /home/thomas/workspace/chainlink/core/services/subscription.go:97 +0x71
github.com/smartcontractkit/chainlink/core/services.StartJobSubscription(0xc0007ee530, 0x1e7e2ed0, 0xed6dd24e6, 0xc0001aa7e0, 0xc0110baa80, 0x1, 0x1, 0x0, 0xc011080c00, 0x5, ...)
        /home/thomas/workspace/chainlink/core/services/subscription.go:53 +0x3f6
github.com/smartcontractkit/chainlink/core/services.(*jobSubscriber).AddJob(0xc0004d1400, 0xc0007ee530, 0x1e7e2ed0, 0xed6dd24e6, 0xc0001aa7e0, 0xc0110baa80, 0x1, 0x1, 0x0, 0xc011080c00, ...)
        /home/thomas/workspace/chainlink/core/services/job_subscriber.go:98 +0xe5
github.com/smartcontractkit/chainlink/core/services.(*jobSubscriber).Connect.func1(0xc0110a05a0, 0xc0110a05a0)
        /home/thomas/workspace/chainlink/core/services/job_subscriber.go:147 +0xc5
github.com/smartcontractkit/chainlink/core/store/orm.(*ORM).Jobs.func1(0x0, 0x3e8, 0x0, 0x7fa49009a228, 0xc010fef600)
        /home/thomas/workspace/chainlink/core/store/orm/orm.go:454 +0x46f
github.com/smartcontractkit/chainlink/core/store/orm.Batch(0x64, 0xc010fe3ad0, 0x138d480, 0xc0007f60a0)
        /home/thomas/workspace/chainlink/core/store/orm/orm.go:1617 +0x4d
github.com/smartcontractkit/chainlink/core/store/orm.(*ORM).Jobs(0xc0006ee880, 0xc010fe3b58, 0xc0110824b0, 0x3, 0x3, 0x0, 0xc0003de000)
        /home/thomas/workspace/chainlink/core/store/orm/orm.go:424 +0x99
github.com/smartcontractkit/chainlink/core/services.(*jobSubscriber).Connect(0xc0004d1400, 0xc0000e8c60, 0x0, 0x0)
        /home/thomas/workspace/chainlink/core/services/job_subscriber.go:145 +0xe5
github.com/smartcontractkit/chainlink/core/services.(*HeadTracker).connect(0xc000654c80, 0xc0000e8c60)
        /home/thomas/workspace/chainlink/core/services/head_tracker.go:236 +0x75
github.com/smartcontractkit/chainlink/core/services.(*HeadTracker).subscribeToHead(0xc000654c80, 0x0, 0x0)
        /home/thomas/workspace/chainlink/core/services/head_tracker.go:554 +0x26d
github.com/smartcontractkit/chainlink/core/services.(*HeadTracker).subscribe(0xc000654c80, 0x1399047)
        /home/thomas/workspace/chainlink/core/services/head_tracker.go:282 +0x3bb
github.com/smartcontractkit/chainlink/core/services.(*HeadTracker).listenForNewHeads(0xc000654c80)
        /home/thomas/workspace/chainlink/core/services/head_tracker.go:254 +0x136
created by github.com/smartcontractkit/chainlink/core/services.(*HeadTracker).Start
        /home/thomas/workspace/chainlink/core/services/head_tracker.go:173 +0x193
```
Using:
```
OPERATOR_CONTRACT_ADDRESS=0x6f86c7e38dA17b72Ac7D8eCBF8979499e4101467
```

**Comment by j16r** [comment]
Should be fixed @thodges-gh

**Review by thodges-gh — APPROVED** [review]
Acceptance completed on Kovan.

**Pull request merged**